common/src/main/scala/explore/syntax/ui/package.scala

```diff
@@ -66,3 +68,21 @@ extension [F[_]: ApplicativeThrow: ToastCtx, A](f: F[A])
         ToastCtx[F]
           .showToast(throwable.getMessage, Message.Severity.Error, sticky = true)
     }
+
+/**
+ * Switch the value of a ViewF to true-ish (by the function) while executing the given effect, then
+ * switch it back to false when the effect is finished
+ */
+extension [F[_]: Monad, A](adding: ViewF[F, A])
+  def useBoolSwitchBy[B](makeBool: Boolean => A)(f: F[B])(using MonadCancel[F, Throwable]): F[B] =
```

Comment:
This seems to require both `Monad` and `MonadCancel`, which is redundant. Maybe you can just do `[F[_]: MonadThrow]` ?